trio/_threads.py

```diff
@@ -215,6 +227,34 @@ def abort(_):
     return await trio.lowlevel.wait_task_rescheduled(abort)
 
 
+# inspired by code from Kovid Goyal https://bugs.python.org/issue15500#msg230736
+def _set_thread_name(name: str):
```

Comment:
This will be a little fiddly to plumb through, but we can do better: because Trio ultimately uses `threading.Thread` under the hood, we can assign to the `.name` attribute which should work on all platforms.  Following `start_thread_soon` through will get you to `trio/_core/_thread_cache.py`.

Additional complication: we actually have a cached pool of threads, so we'll want to set the name when we starting working, and reset to a generic e.g. `trio-idlethread` when the task is over.

Reply:
could avoid having to plumb through several levels of call by using `threading.current_thread()` instead of `.get_ident()`.

That doesn't sound to bad, just unset the name after `sync_fn(*args)` is called in `worker_fn`? Though I guess that might fit better in `_core/_thread_cache` and is an argument for doing proper plumbing and placing all the logic in there.

Reply:
Yeah, I'd definitely put it in https://github.com/python-trio/trio/blob/ce4c99e696a516db40b55fb076deca2e586626cf/trio/_core/_thread_cache.py#L59-L70

Then we can just do `self.thread.name = thread_name` before `result = outcome.capture(fn)` and reset afterwards.  It also doesn't look like this interface shares the 15-character limit?

Reply:
yeah the 15-char limit is just in pthreads / the OS, thread.name is only used internally in python so that's a normal unicode string.